packages/core/src/dispatcher.ts

```diff
@@ -345,21 +345,27 @@ export class Dispatcher {
    */
   async _handleResponseMessage(message: ResponseMessage): Promise<void> {
     const { id: queryId, tips } = message
-    const expectedStreamID = this.messageBus.outstandingQueries.get(queryId)
-    if (expectedStreamID) {
-      const newTip = tips.get(expectedStreamID.toString())
-      if (!newTip) {
-        throw new Error(
-          "Response to query with ID '" +
-            queryId +
-            "' is missing expected new tip for StreamID '" +
-            expectedStreamID +
-            "'"
-        )
-      }
-      this.repository.stateManager.update(expectedStreamID, newTip)
-      this.messageBus.outstandingQueries.delete(queryId)
-      // TODO Iterate over all streams in 'tips' object and process the new tip for each
+    const outstandingQuery = this.messageBus.outstandingQueries.queryMap.get(queryId)
+    const expectedStreamID = outstandingQuery?.streamID
+    const newTip = tips.get(expectedStreamID.toString())
+    if (!newTip) {
+      throw new Error(
+        "Response to query with ID '" +
+          queryId +
+          "' is missing expected new tip for StreamID '" +
+          expectedStreamID +
+          "'"
+      )
+    }
+    this.repository.stateManager.update(expectedStreamID, newTip)
+    // TODO Iterate over all streams in 'tips' object and process the new tip for each
+
+    //either way, cleanup outstanding queries
+    try {
+      this.messageBus.outstandingQueries.cleanUpExpiredQueries()
```

Comment:
I think this should happen automatically inside the outstandingQueries object every time a new query is added.  That way the garbage collection can be entirely an implementation detail of the OutstandingQueries class, and calling code doesn't need to know anything about it, or have to remember to call anything to make sure it happens.